### PR TITLE
Create bot within class method to encapsulate all bot initialization logic

### PR DESCRIPTION
**Pull request by DudeBro249**
Using a python classmethod `create` within the `ArenaBot` class to create the bot and encapsulate all bot initialization logic. Also changed `README.md` to reflect correct bot setup instructions according to this pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,21 +47,25 @@ python -m api
...
 ```
 TOKEN=
 API_URL=
 ```
-3. Run the bot with
+4. Run the bot with
 ```cmd
-python -m bot
+python __main__.py
 ```
```
Comment (GDWR) [inline README.md:69]: I would expect the bot to be run as a module and the documentation should definitely show this.

```cmd
Python -m bot
```

While in src/

**Review by GDWR — CHANGES_REQUESTED** [review]
Review comment about `README.md` then after it is resolved this will be merged.

File: README.md (hunk 1 above)
Comment (GDWR) [reply README.md:69]: while being in the directory and using `__main__.py` should basically be the same,
I would prefer the cleaner way and the intended way to be shown in the README

**Comment by GDWR** [comment]
I have just added linting to any new Pull requests as an action. 
As this was made before had the trigger wasn't triggered. 

Please can make sure that your changes meet our flake8 requirements by running
`pip3 install flake8`
`flake8 src/api`
`flake8 src/bot`

  || I think `flake8 .` would also work but I prefer separating them ||
There has already been some changes made to meet flake8 in `main`

File: README.md (hunk 1 above)
Comment (DudeBro249) [reply README.md:69]: Sure, I'll revert those changes.

Comment (DudeBro249) [reply README.md]: I've reverted the changes, please go through the changes whenever possible.

File: src/bot/arena_bot.py [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
 
     async def on_ready(self):
         print(f"Logged in as {self.user}")
-    
+
     async def on_disconnect(self):
         await self.session.close()
+
+    @classmethod
+    def create(cls) -> 'ArenaBot':
```
Comment (GDWR) [inline src/bot/arena_bot.py:19]: Can we use
```py
from __future__ import annotations
```
to allow annotating returns before class definition?

**Review by GDWR — CHANGES_REQUESTED** [review]

File: src/bot/arena_bot.py (hunk 2 above)
Comment (DudeBro249) [reply src/bot/arena_bot.py:19]: Sure, I'll take a look at how to do that.

**Pull request closed without merging**